.github/.check_scopes.lua

```diff
@@ -0,0 +1,38 @@
+function extractScopesFromFile(filename)
+    local scopes = {}
+    local file = io.open(filename, "r")
+    if not file then
+        error("Failed to open file: " .. filename)
+    end
+    for line in file:lines() do
+        local scope = line:match("%S+")
+        if scope then
+        scopes[scope] = true
+        end
+    end
+    file:close()
+    return scopes
+end
+
+function extractScopeFromCommit(commitTitle)
+    local scope = commitTitle:match("%a+%((.-)%)")
+    return scope
```

Comment:
This can be simplified

```suggestion
    return commitTitle:match("%a+%((.-)%)")
```